CMakeLists.txt

```diff
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.9)
 project("llama.cpp" C CXX)
 
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
I am not a fan of setting this globally. this should be specified for each target.
`target_compile_features(llama PUBLIC cxx_std_17)` or similar. same for c.

always consider what happens when the project is included via `add_subdirectoy()`